### PR TITLE
NMS-16163: Update confd for new datasources XML schema

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16163

### CODE REVIEW

**Comment by mmahacek** [comment]
Docs for SSL config will be part of #6692

File: opennms-container/core/container-fs/confd/templates/opennms-datasources.xml.tpl [hunk 1]
```diff
@@ -7,25 +7,41 @@
...
-    minPool="{{getv "/opennms/database/connection/minpool" "50"}}"
+    minPool="{{getv "/opennms/database/connection/minpool" "25"}}"
     maxPool="{{getv "/opennms/database/connection/maxpool" "50"}}"
     maxSize="{{getv "/opennms/database/connection/maxsize" "50"}}" />
 
   <jdbc-data-source name="opennms"
                     database-name="{{getv "/opennms/dbname" "opennms"}}"
                     class-name="org.postgresql.Driver"
-                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/{{getv "/opennms/dbname" "opennms"}}"
+                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/{{getv "/opennms/dbname" "opennms"}}?sslmode={{getv "/postgres/ssl/mode" "disable"}}&amp;sslfactory={{getv "/postgres/ssl/factory" "org.postgresql.ssl.NonValidatingFactory"}}"
```
Comment (mmahacek) [inline opennms-container/core/container-fs/confd/templates/opennms-datasources.xml.tpl:20]:
```suggestion
                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/{{getv "/opennms/dbname" "opennms"}}?sslmode={{getv "/postgres/ssl/mode" "disable"}}&amp;sslfactory={{getv "/postgres/ssl/factory" "org.postgresql.ssl.LibPQFactory"}}"
```

File: opennms-container/core/container-fs/confd/templates/opennms-datasources.xml.tpl [hunk 2]
```diff
@@ -7,25 +7,41 @@
...
                     user-name="{{getv "/opennms/dbuser" "opennms"}}"
                     password="{{getv "/opennms/dbpass" "opennms"}}" />
 
   <jdbc-data-source name="opennms-admin"
                     database-name="template1"
                     class-name="org.postgresql.Driver"
-                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/template1"
+                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/template1?sslmode={{getv "/postgres/ssl/mode" "disable"}}&amp;sslfactory={{getv "/postgres/ssl/factory" "org.postgresql.ssl.NonValidatingFactory"}}"
```
Comment (mmahacek) [inline opennms-container/core/container-fs/confd/templates/opennms-datasources.xml.tpl:27]:
```suggestion
                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/template1?sslmode={{getv "/postgres/ssl/mode" "disable"}}&amp;sslfactory={{getv "/postgres/ssl/factory" "org.postgresql.ssl.LibPQFactory"}}"
```

File: opennms-container/core/container-fs/confd/templates/opennms-datasources.xml.tpl [hunk 3]
```diff
@@ -7,25 +7,41 @@
...
+    <connection-pool idleTimeout="{{getv "/opennms/database/connection/idletimeout" "600"}}"
+                     minPool="{{getv "/opennms/database/connection/minpool" "0"}}"
+                     maxPool="{{getv "/opennms/database/connection/maxpool" "10"}}"
+                     maxSize="{{getv "/opennms/database/connection/maxsize" "50"}}" />
+  </jdbc-data-source>
+
+  <jdbc-data-source name="opennms-monitor"
+                    database-name="postgres"
+                    class-name="org.postgresql.Driver"
+                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/postgres?sslmode={{getv "/postgres/ssl/mode" "disable"}}&amp;sslfactory={{getv "/postgres/ssl/factory" "org.postgresql.ssl.NonValidatingFactory"}}"
```
Comment (mmahacek) [inline opennms-container/core/container-fs/confd/templates/opennms-datasources.xml.tpl:39]:
```suggestion
                    url="jdbc:postgresql://{{getv "/postgres/host" "database"}}:{{getv "/postgres/port" "5432"}}/postgres?sslmode={{getv "/postgres/ssl/mode" "disable"}}&amp;sslfactory={{getv "/postgres/ssl/factory" "org.postgresql.ssl.LibPQFactory"}}"
```

**Review by RangerRick — APPROVED** [review]
I assume it's "legal" to use sub-sub-keys without them being explicitly defined?

**Comment by mmahacek** [comment]
It's not _illegal_ and has been working for my tests...

**Pull request merged**